pkg/operators/process/helpers/processhelpers_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2025 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processhelpers_test
+
+import (
+	"os"
+	"runtime"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	processhelpers "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/process/helpers"
+)
+
+type allOptions struct {
+	totalMemory uint64
+	cpuCount    int
+}
+
+func (a allOptions) WithCPUUsage() bool {
+	return true
+}
+
+func (a allOptions) WithCPUUsageRelative() bool {
+	return true
+}
+
+func (a allOptions) WithComm() bool {
+	return true
+}
+
+func (a allOptions) WithPPID() bool {
+	return true
+}
+
+func (a allOptions) WithState() bool {
+	return true
+}
+
+func (a allOptions) WithUID() bool {
+	return true
+}
+
+func (a allOptions) WithVmSize() bool {
+	return true
+}
+
+func (a allOptions) WithVmRSS() bool {
+	return true
+}
+
+func (a allOptions) WithMemoryRelative() bool {
+	return true
+}
+
+func (a allOptions) WithThreadCount() bool {
+	return true
+}
+
+func (a allOptions) WithStartTime() bool {
+	return true
+}
+
+func (a allOptions) TotalMemory() uint64 {
+	return a.totalMemory
+}
+
+func (a allOptions) NumCPU() int {
+	return a.cpuCount
+}
+
+func (a allOptions) LastCPUTime(pid int) (uint64, bool) {
+	return 0, false
+}
+
+func (a allOptions) BootTime() time.Time {
+	return time.Now()
+}
+
+func TestGetProcessInfo(t *testing.T) {
+	pid := os.Getpid()
+	proc, err := processhelpers.GetProcessInfo(pid, 0, allOptions{
+		cpuCount: 1,
+	})
+	require.NoError(t, err)
+
+	assert.Equal(t, pid, proc.PID)
+	assert.Greater(t, proc.PPID, 0)
+	assert.NotEmpty(t, proc.Command)
+	assert.Zero(t, proc.CPUUsage)
```

Comment:
Why CPU usage is 0 for this process?

Reply:
It can only get the CPU usage on second run (where it has the delta and can calculate it from that).